pkg/devices/memballoon.go

```diff
@@ -0,0 +1,47 @@
+package devices
+
+type MemballoonModel int
+
+const (
+	MemBalloonModelNone MemballoonModel = iota
+	MemBalloonModelVirtio
+	MemBalloonModelVirtioNonTransitional
+	MemBalloonModelVirtioTransitional
+)
+
+type Memballoon struct {
+	model MemballoonModel
+}
+
+func (self MemballoonModel) String() string {
```

Comment:
Using `String()` here instead of `ToStr`, since `String()` will be automatically picked up by `fmt.Print()`. Also note that you can also attach methods on types like `MemballoonModel`. I guess in you rexample it is important to have the `int` values as base to allow seamless integration with libvirt? If really the `string` value is the important bit, then you can also consider making `type MemballoonModel string`.

Reply:
> Using `String()` here instead of `ToStr`, since `String()` will be automatically picked up by `fmt.Print()`.

Makes sense, I imaged there would be some conventional name for this functionality but forgot to research it :)

> Also note that you can also attach methods on types like `MemballoonModel`.

That's pretty nice!

> I guess in you rexample it is important to have the `int` values as base to allow seamless integration with libvirt? If really the `string` value is the important bit, then you can also consider making `type MemballoonModel string`.

Correct, the use of `int` is for interoperability with C. Perhaps we could be more clever here but it seems acceptable for the time being.